go-controller/pkg/ovn/address_set/address_set.go

```diff
@@ -66,43 +74,87 @@ type AddressSet interface {
 }
 
 type ovnAddressSetFactory struct {
-	nbClient libovsdbclient.Client
+	nbClient       libovsdbclient.Client
+	controllerName string
+}
+
+// DbIndex is the copy of libovsdbops.DBIndex, but without OwnerController field, since it
```

Comment:
Yeah I agree this is a bit ugly, so my concern is that we need to make sure that only objects owned by a specific controller are modified by that controller, if someone just forgets to set `OwnerController` in the future, it can modify the wrong object.
So I tried to make sure we don't just pass DbIndex with empty `OwnerController` hoping someone else will set this field, but explicitly use another type for this.
I am open to any ideas on how to make that prettier :)

Reply:
We talked on slack about putting comments on dbIndex to indicate how it should be used, and having validation methods used in dbIndex to santize someone trying to set external_ids to make sure the required ones are filled out. It is up to the controller implementations to enforce that they are the owner when using dbIndexes. So for example addressSet or addressSet factory would override any user supplied owner and enter itself, and then call the validation dbIndex method before trying to write external_ids to OVN. Developers should be aware of controller ownership and how it should be properly used.